### PR TITLE
DOCSP-9597 delete one

### PR DESCRIPTION
**Pull request by ccho-mongodb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-9597

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/4d2fd18/java/docsworker-xlarge/DOCSP-9597-delete-one/usage-examples/deleteOne

### CODE REVIEW

File: source/usage-examples/deleteOne.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+filter to match movies with the ``title`` exactly matching the text
+``'The Garbage Pail Kids Movie'``.
+
+.. include:: /includes/connect-guide-note.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/deleteOne.java
+   :language: java
+
+If you are using the **legacy API**, see the :doc:`FAQ </faq>` to determine 
+what changes you need to make to this code example.
```
Comment (nlarew) [inline source/usage-examples/deleteOne.txt:35]: Should we put this in a note? Seems irrelevant for most users so we might wanna set it off visually.

File: source/usage-examples/deleteOne.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+a filter, MongoDB matches the first document in the collection. The 
+``deleteOne()`` method only deletes the first document matched.
+
+This method returns an instance of ``DeleteResult`` which contains information 
+including how many documents were deleted as a result of the operation.
+
+If your delete operation fails, the driver raises an exception. For more
+information on the types of exceptions raised under specific conditions,
+see the API documentation for ``deleteOne()``, linked at the bottom of
+this page.
```
Comment (nlarew) [inline source/usage-examples/deleteOne.txt:19]: Why not just use inline links to `deleteOne()` and `DeleteResult`? Feels a little annoying to have to scroll down to the bottom.

File: source/includes/usage-examples/code-snippets/deleteOne.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        String uri = "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+
+        try (MongoClient mongoClient = MongoClients.create(uri)) {
+
+            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
+            MongoCollection<Document> collection = database.getCollection("movies");
+
+            Bson query = eq("title", "The Garbage Pail Kids Movie");
+
+            System.out.println(collection.find(query).first());
```
Comment (nlarew) [inline source/includes/usage-examples/code-snippets/deleteOne.java:28]: Maybe add some descriptive text to this `println`? e.g. "Document to delete: { ... }"

Comment (ccho-mongodb) [reply source/includes/usage-examples/code-snippets/deleteOne.java:28]: Whoops, this was for debugging, will remove.

File: source/usage-examples/deleteOne.txt (hunk 2 above)
Comment (ccho-mongodb) [reply source/usage-examples/deleteOne.txt:19]: This is in response to some feedback from drivers teams that the inline links pulled them away from the content of the page too early. I can revisit this again at some point if it seems like this is makes it difficult for the docs to use.

File: source/usage-examples/deleteOne.txt (hunk 1 above)
Comment (ccho-mongodb) [reply source/usage-examples/deleteOne.txt:35]: Sure. I was undecided whether this belongs in a note since I always thought notes were visual cues that drew the eye which would be placing more importance on that information rather than less.
